controller/controller.go

```diff
@@ -220,14 +220,33 @@ func (c *Controller) RunOnce(ctx context.Context) error {
 		ManagedRecords:     c.ManagedRecordTypes,
 	}
 
-	plan = plan.Calculate()
+	p = p.Calculate()
 
-	if plan.Changes.HasChanges() {
-		err = c.Registry.ApplyChanges(ctx, plan.Changes)
+	if p.Changes.HasChanges() {
+		err = c.Registry.ApplyChanges(ctx, p.Changes)
 		if err != nil {
-			registryErrorsTotal.Inc()
-			deprecatedRegistryErrors.Inc()
-			return err
+			// if the original batch fails, try each action
+			createOnlyPolicy := &plan.CreateOnlyPolicy{}
+			createChanges := createOnlyPolicy.Apply(p.Changes)
+			createErr := c.Registry.ApplyChanges(ctx, createChanges)
+			// if the single action batch fails, try batch in halves
+			if createErr != nil {
+				applyHalfActionChanges(c, ctx, createChanges)
```

Comment:
Since `applyHalfActionChanges` returns an error we should get it here and at least log it.